### ISSUE TITLE
frequent crashes

### ISSUE THREAD

**Issue by jonorthwash** [linuxmint/muffin#583]
```
 * muffin version 4.8.1
 * Distribution - Debian sid
 * Graphics hardware: Intel(R) HD Graphics 620 (KBL GT2)
 * Drivers loaded: glamoregl, DRI driver: iris, VDPAU driver: va_gl
 * 64 bit
```

**Issue**

I often leave my computer for several hours and return to find that cinnamon/muffin has crashed.  The logs report the following:

```
failed to create drawable
failed to create drawable
failed to create drawable
[xcb] Unknown sequence number while processing queue
[xcb] Most likely this is a multi-threaded client and XInitThreads has not been called
[xcb] Aborting, sorry about that.
cinnamon: ../../src/xcb_io.c:269: poll_for_event: Assertion `!xcb_xlib_threads_sequence_lost' failed.
Aborted.
```

`cinnamon --replace` gets things back up and running again fine, but often for only about a day before it happens again.

**Steps to reproduce**

Not sure.  I leave my computer for a while and come back to this.

**Expected behaviour**

Cinnamon/muffin runs indefinitely without crashing.

**Other information**

This started around when I reenabled my screensaver after having disabled it for a few days.  It may have something to do with that?

This may be a cinnamon-specific issue and not related to muffin.  If so advised, I am happy to repost in the appropriate spot.

**Comment by tycho-kirchner** [comment]
It sounds very similar to [this](https://gitlab.freedesktop.org/xorg/lib/libx11/-/merge_requests/34) bug - please check if your version of libx11 already includes the fix. To check, if its really cinnamons fault, try to reproduce in another DE (mate, xfce4 ...). See also [here](https://bugs.launchpad.net/ubuntu/+source/libx11/+bug/1782984).
Good luck!

**Comment by ClundXIII** [comment]
I can confirm this bug on `Linux Mint 20.2 Uma Cinnamon`. (Although it is slightly different, it is inside `xcb_io.c:260`)

Unfortunately ubuntu 20.04 repos still only have `libx11 1.6.9`. Do you know if there is any plan to update the package? Or do I have to manually upgrade it?

**Comment by tycho-kirchner** [comment]
Looking at the [changelog](http://changelogs.ubuntu.com/changelogs/pool/main/libx/libx11/libx11_1.6.9-2ubuntu1.2/changelog) I don't think the fix has made it's way into the repo. Personally I have done so manually already months ago ...

**Comment by gkatev** [comment]
Hi,
I'm also encountering this bug, with the exact same message as the OP (found in .xsession-errors).

For me it just happened twice when unlocking (not logging into) my system, and I've also been having some crashes for some days now. I also specifically remember some of them occuring when I open some tray icons' menus, but I have not confirmed if it is the same error.

I'm on Arch Linux, with cinnamon 5.0.5, muffin 5.0.1, libx11 1.7.2. AMD Navi 10 GPU, with amdgpu driver.

**Comment by ClundXIII** [comment]
In my case the crash is often triggered by opening or closing `guake`.

**Comment by tycho-kirchner** [comment]
> I'm on Arch Linux, with cinnamon 5.0.5, muffin 5.0.1, libx11 1.7.2

Mhm, your version of libx11 should be new enough, maybe it's another issue?
On the other hand: I do not run the final version of the above mentioned merge request but an earlier one (Version 5). Maybe the later versions again introduced the regression (more seldomly)? With [this](https://gitlab.freedesktop.org/xorg/lib/libx11/-/merge_requests/34/diffs?diff_id=61521) patch I did not experience the bug for moths ...

**Comment by leigh123linux** [comment]
Report the issue against the graphic driver.

**Comment by gkatev** [comment]
Posting for completeness's sake, even though the issue's been closed:

> > I'm on Arch Linux, with cinnamon 5.0.5, muffin 5.0.1, libx11 1.7.2
> 
> Mhm, your version of libx11 should be new enough, maybe it's another issue?
> On the other hand: I do not run the final version of the above mentioned merge request but an earlier one (Version 5). Maybe the later versions again introduced the regression (more seldomly)? With [this](https://gitlab.freedesktop.org/xorg/lib/libx11/-/merge_requests/34/diffs?diff_id=61521) patch I did not experience the bug for moths ...

This is the output I see on .xsession-errors, don't know if the WARNINGs are relevant (probably not).
```
cinnamon-session[1256]: WARNING: t+76882.99660s: Detected that screensaver has appeared on the bus

(cinnamon:85104): Cogl-WARNING **: 15:22:20.284: Skipping layers 1..n of your pipeline since the first layer is sliced. We don't currently support any multi-texturing with sliced textures but assume layer 0 is the most important to keep
[xcb] Unknown sequence number while processing queue
[xcb] Most likely this is a multi-threaded client and XInitThreads has not been called
[xcb] Aborting, sorry about that.
cinnamon: xcb_io.c:269: poll_for_event: Assertion `!xcb_xlib_threads_sequence_lost' failed.
```

I will add to my to-do list (!) to look around, regarding the libx11 version and the mentioned patch, and who to report this problem to. Definitely do feel free to beat me to it though :)

**Comment by ClundXIII** [comment]
> Report the issue against the graphic driver.

Unfortunately I cannot easily use the fixed version of the library. Even if I manually install the 1.7 version of `libx11-xcb -data -6 -dev` etc I run into the Problem that `libx11-6:i386` depends on `libc6 >= 2.33`.

I think the best solution would be noticing the ubuntu maintainers to update libx11 to 1.7 (and maybe fixing the i386 build so it does not require a libc upgrade)

**Comment by tycho-kirchner** [comment]
@ClundXIII 
I think instead of posting to this closed issue, please report it here:

https://bugs.launchpad.net/ubuntu/+source/libx11/+bug/1782984

Btw. looking at the source-package of libx11-6, ubuntu 20.04 the bug is **not fixed** (search for 'next_response' in src/Xxcbint.h).

And I think the best solution would be to backport the short enough fix to 1.6.9 instead of upgrading to 1.7 - that's what I did and I'm running fine with that for months.

**Comment by gkatev** [comment]
FYI, my comment on the X11 issue: https://gitlab.freedesktop.org/xorg/lib/libx11/-/merge_requests/34#note_1054538
@ClundXIII also reports the problem persisting here: https://bugs.launchpad.net/ubuntu/+source/libx11/+bug/1782984/comments/110, as well as some connection with the desktop resolution.

In what way might something like this be connected to the X-screen's resolution? (I have not yet tried lowering mine). Two weird resolution-related issues that spring to mind:
https://gitlab.gnome.org/GNOME/gnome-screenshot/-/issues/115
https://gitlab.gnome.org/GNOME/gnome-screenshot/-/issues/168
May be wildly irrelevant, may not be.

Am I going about this the wrong way? Is the X11 issue unrelated or might it not still be fixed? Is the bug I come upon due to muffin? Or should I be reporting this on mesa's issue tracker, instead of x11's and muffin's (it will be my next step :) )?
Any input welcome.
